scipy/special/_logsumexp.py

```diff
@@ -172,11 +171,10 @@ def _elements_and_indices_with_max_real(a, axis=-1, xp=None):
         # simple, array-API compatible way of doing so that doesn't
         # have a problem with `axis` being a tuple or None.
         i = xp.reshape(xp.arange(xp_size(a)), a.shape)
-        i[~mask] = -1
+        i = xpx.at(i, ~mask).set(-1)
```

Comment:
This fails on jax.jit. My current intention is to change jax.jit itself to special-case `arr.at[idx].set(value)` when idx is a boolean mask and value is a scalar, so that it can be rewritten as `jnp.where(idx, value, arr)`. Failing that, I can implement the same special case in array-api-extra.